### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Most notable: `github.com/hashicorp/golang-lru` to `/v2`

### CODE REVIEW

File: pubsub/pulsar/pulsar.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ import (
 	"time"
 
 	"github.com/apache/pulsar-client-go/pulsar"
-	lru "github.com/hashicorp/golang-lru"
+	lru "github.com/hashicorp/golang-lru/v2"
```
Comment (ItalyPaleAle) [inline pubsub/pulsar/pulsar.go:25]: Reviewers: this file contains some actual (small) code changes since the v2 of the library uses generics

**Review by berndverst — DISMISSED** [review]
Please redo this. Never manually update indirect dependencies. It causes subtle bugs.

Just update direct dependencies and let the indirect ones resolve automatically.

**Comment by berndverst** [comment]
Let's wait for the tests to pass. Then I will merge master and force merge this!

**Pull request merged**